src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTestBase.cs

```diff
@@ -145,27 +145,48 @@ internal static EntityHandle Handle(int rowNumber, TableIndex table)
         internal static bool IsDefinition(HandleKind kind)
             => kind is not (HandleKind.AssemblyReference or HandleKind.ModuleReference or HandleKind.TypeReference or HandleKind.MemberReference or HandleKind.TypeSpecification or HandleKind.MethodSpecification);
 
+        /// <summary>
+        /// Checks that the EncLog contains specified rows.
+        /// Any default values in the expected <paramref name="rows"/> are ignored to facilitate conditional code.
+        /// </summary>
         internal static void CheckEncLog(MetadataReader reader, params EditAndContinueLogEntry[] rows)
         {
-            AssertEx.Equal(rows, reader.GetEditAndContinueLogEntries(), itemInspector: EncLogRowToString);
+            AssertEx.Equal(
+                rows.Where(r => r.Handle != default),
```

Comment:
I didn't follow the purpose of this change. Was it necessary?

Reply:
The test conditionally adds rows and needs the default ones be ignored.